### PR TITLE
Fix/builder duping

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Stops the builder from requesting the double. Should test this before merging.
- Less influence of security on happiness
- Some other bugs
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/ColonyView.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
      */
     private Set<Block> freeBlocks = new HashSet<>();
 
-    private double overallHappiness;
+    private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyView.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockHutWareHouse, 1), "XXX", "X#X", "XXX", 'X', PLANK_WOOD, '#', "chest")); // check
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockHutDeliveryman, 1), "XXX", "X#X", "XXX", 'X', PLANK_WOOD, '#', Items.LEATHER_BOOTS));
 
-        enableInDevelopmentFeatures(enableInDevelopmentFeatures);
+        //enableInDevelopmentFeatures(enableInDevelopmentFeatures);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 46 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 40 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyView.java#L27](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L27): class "ColonyView" has 42 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyView.java#L93](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L93): Remove this unused method parameter "isNewSubScription". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [ColonyView.java#L354](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L354): Provide multiple methods instead of using "isNewSubscription" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L182](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L182): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L82](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L82): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/5c04c86bfb879ea1d90a5d00e8c0d8ce77240806/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
@Raycoms tell me once you tested

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 49 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyView.java#L27](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L27): class "ColonyView" has 42 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyView.java#L93](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L93): Remove this unused method parameter "isNewSubScription". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [ColonyView.java#L354](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L354): Provide multiple methods instead of using "isNewSubscription" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L182](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L182): The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L82](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L82): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [StructureWrapper.java#L25](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/util/StructureWrapper.java#L25): class "StructureWrapper" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [StructureWrapper.java#L196](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/coremod/util/StructureWrapper.java#L196): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/cfacc8a24fb655778b9b349f32e12e51cd457546/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 67 issues
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyView.java#L27](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L27): class "ColonyView" has 42 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyView.java#L93](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L93): Remove this unused method parameter "isNewSubScription". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [ColonyView.java#L354](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L354): Provide multiple methods instead of using "isNewSubscription" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 64 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): class "EntityCitizen" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): Refactor this class so it has no more than 20 fields, rather than the 21 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/347c7a4c1fb43604f3e1e9497c2a89de78a65cfc/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Assign this magic number 0.02F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 67 issues
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyView.java#L27](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L27): class "ColonyView" has 42 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyView.java#L93](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L93): Remove this unused method parameter "isNewSubScription". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [ColonyView.java#L354](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/colony/ColonyView.java#L354): Provide multiple methods instead of using "isNewSubscription" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 64 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): class "EntityCitizen" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L66](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L66): Refactor this class so it has no more than 20 fields, rather than the 21 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/80458768cbcbbe9d14d6067d1e5ff3e09daea41f/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Assign this magic number 0.02F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 83 issues
* ![MAJOR][MAJOR] 37 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.201 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 42 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): class "Colony" has 81 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Refactor this class so it has no more than 20 fields, rather than the 35 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L172](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L172): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L547](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L547): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1007](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1007): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1008](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1008): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1131](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1131): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1133](https://github.com/Minecolonies/minecolonies/blob/0c3f9ac60a4b44314553d619cca3d9bb9e0531af/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1133): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 77 issues
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.201 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 42 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): class "Colony" has 81 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Refactor this class so it has no more than 20 fields, rather than the 35 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L172](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L172): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L547](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L547): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1007](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1007): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1008](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1008): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1131](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1131): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1133](https://github.com/Minecolonies/minecolonies/blob/61efcf8ff080fcc58761ea86e20b2b3e1f17a780/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1133): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/984/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**